src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2995,7 +2995,11 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
                 var binder = binderFactory.GetBinder(paramList);
 
                 var ctor = addCtor(builder.RecordDeclarationWithParameters);
-                addProperties(ctor.Parameters);
+                var existingOrAddedMembers = addProperties(ctor.Parameters);
+                if (existingOrAddedMembers.Length == ctor.Parameters.Length)
```

Comment:
>existingOrAddedMembers.Length == ctor.Parameters.Length [](start = 20, length = 55)

This test is not correct.  According to the spec at [https://github.com/dotnet/csharplang/blob/master/proposals/records.md#deconstruct](https://github.com/dotnet/csharplang/blob/master/proposals/records.md#deconstruct) the deconstructor should be produced whether or not the properties are synthesized.

Reply:
Removed the test.